pes-sa-remove/pes-sa-remove.php

```diff
@@ -0,0 +1,141 @@
+<?php
+### Copyright 1999-2024. WebPros International GmbH.
+
+###############################################################################
+# This script shows addresses in spamfilter and spamfilter_preference that are not present in Plesk.
+# Requirements : >php7.4, GNU coreutils
+# Version      : 1.1
+#########
+$short_options = "fh";
+$long_options = ["fix","help"];
+$options = getopt($short_options, $long_options);
+
+$usage = "Here is the list of the available options: \n
+-f, --fix      If this option is defined the affected mailboxes will be removed from the SpamAssassin table;
+-h, --help     Show the script usage;\n";
+
+if (isset($options["h"]) || isset($options["help"])) {
+    print($usage);
+    exit;
+}
+
+function SqlQuery($query){
+    //Function for calling MySQL
+
+    $mysql_pass = trim(file_get_contents('/etc/psa/.psa.shadow', false));
+    mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT);
+    $mysql = new mysqli('localhost', 'admin', $mysql_pass, 'psa');
+
+    $q_result = mysqli_query($mysql,$query);
+
+    return $q_result;
+}
+
+function RemoveRecords($box_id){
+    //Removing the incorrect records from the SpamFilter table
+
+    $r_pref_query = "delete from spamfilter_preferences where spamfilter_id = $box_id";
```

Comment:
There are a lot of potential SQL injections. Prepared statements should be used instead.